### PR TITLE
GEODE-7802: Add disclaimer for OQL queries in UG

### PR DESCRIPTION
**Pull request by jvarenina**
Disclaimer: OQL queries are not pre-validated for nonexistent attributes
and methods

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-docs/developing/querying_basics/restrictions_and_unsupported_features.html.md.erb [hunk 1]
```diff
@@ -30,6 +30,6 @@ At a high level, <%=vars.product_name%> does not support the following querying
 
 -   You cannot create an index on fields using Set/List types (Collection types) that are not comparable. The OQL index implementation expects fields to be Comparable. To workaround this, you can create a custom Collection type that implements Comparable.
 -   ORDER BY is only supported with DISTINCT queries.
-
+-   OQL queries aren't pre-validated for references to nonexistent methods and attributes.
```
Comment (agingade) [inline geode-docs/developing/querying_basics/restrictions_and_unsupported_features.html.md.erb:33]: I have commented on the ticket with my thoughts. 
What could be added here is, saying OQL query engine determines the field or methods on the given object (region keys and values) during query execution time.

Comment (davebarnes97) [reply geode-docs/developing/querying_basics/restrictions_and_unsupported_features.html.md.erb:33]: I will propose a rewrite tomorrow that uses @agingade's verbiage.

Comment (davebarnes97) [reply geode-docs/developing/querying_basics/restrictions_and_unsupported_features.html.md.erb:33]: [My suggested revision, adding some of Anil's explanation from the JIRA comments for context, and expanding contractions to whole words for better translation/localization]
In the Geode model, attributes and methods are resolved at runtime during query execution. This means that OQL queries are not pre-validated for references to nonexistent methods and attributes.

Comment (jvarenina) [reply geode-docs/developing/querying_basics/restrictions_and_unsupported_features.html.md.erb]: Thank you both for comments and proposals. I have updated PR based on the @davebarnes97 proposal.

**Review by davebarnes97 — APPROVED** [review]

**Pull request merged**